### DISCUSSION TITLE
STARSolo Input fastq from different samples, but same barcodes

### DISCUSSION THREAD

**Discussion by sgsayson** [alexdobin/STAR#1592] [Q&A] [answered]
When inputting multiple fastq files for R1/R2 (i.e. Sample1_R1.fq, Sample1_R2.fq, Sample2_R1.fq, Sample2, R2.fq, SampleX [...]), does demultiplexing occur per sample? or does it pool it then attempt to demultiplex?

i.e. Barcode XYZ in Sample 1 shows a specific gene profile (lets call this profile ABC). Barcode XYZ in Sample 2 shows profile DEF. 

will it show:
Sample1 XYZ = ABC; Sample2 XYZ = DEF

or will it show:
Barcode XYZ = ABC+DEF

Thanks for any help

**Comment by alexdobin** [comment]
Hi @sgsayson 

multiple comma-separated FASTQs are considered as one file, so demultiplexing happens for all of them.
For separate demultiplexing, you would need to run each one separately.

**Reply by sgsayson** [reply]
Thanks for the reply. 

So for downstream analyses, how would I pool the samples back together, but keep the demultiplexed samples labeled separately? As in, how are they handled in the matrix, gene table, and the barcode table? Any help is greatly appreciated!

**Comment by alexdobin** [comment, marked as answer]
You can add a distinct string to each of the barcodes in the barcodes.tsv file, e.g. _1 for all barcodes in sample 1, _2 for sample 2, etc.
Also, the downstream software (scanpy, seurat) typically allows to load each sample separately, and then concatenate them into one data structure, adding distinct labels for each cell.

**Reply by sgsayson** [reply]
Perfect! Thanks!
